DTML.EduBot/Common/MessageTranslator.cs

```diff
@@ -0,0 +1,101 @@
+﻿namespace DTML.EduBot.Common
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Net.Http;
+    using System.Net.Http.Headers;
+    using System.Configuration;
+    using System.Threading.Tasks;
+    using System.Web;
+    using System.Xml.Linq;
+
+    public class MessageTranslator
+    {
+        /// <summary>
+        /// Get authentication token for translator API
+        /// </summary>
+        /// <param name="key">OcpApimSubscriptionKey</param>
+        /// <returns>token if the request is successful null otherwise</returns>
+        public static async Task<string> GetAuthenticationTokenAsync(string key)
+        {
+            using (var client = new HttpClient())
+            {
+                string token = null;
+                client.DefaultRequestHeaders.Add("Ocp-Apim-Subscription-Key", key);
+                var response = await client.PostAsync(System.Configuration.ConfigurationManager.AppSettings["tokenEndpoint"], null);
+                if (response.IsSuccessStatusCode)
+                {
+                    token = await response.Content.ReadAsStringAsync();
+                }
+                return token;
+            }
+        }
+
+        /// <summary>
+        /// Identify language for input text
+        /// </summary>
+        /// <param name="inputText">text whose language should be detected</param>
+        /// <returns></returns>
+        public static async Task<string> IdentifyLangAsync(string inputText)
+        {
+            string inputTextLang = "en";
+
+            var accessToken = await MessageTranslator.GetAuthenticationTokenAsync(ConfigurationManager.AppSettings["TranslatorApiKey"]);
+            if (accessToken == null)
+            {
+                return inputTextLang;
+            }
+
+            string query = $"?text={System.Net.WebUtility.UrlEncode(inputText)}&contentType=text/plain";
+            using (var client = new HttpClient())
+            {
+                client.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue("Bearer", accessToken);
+                var response = await client.GetAsync(ConfigurationManager.AppSettings["detectLanguageEndpoint"] + query);
+                if (response.IsSuccessStatusCode)
+                {
+                    var result = await response.Content.ReadAsStringAsync();
+                    if (result != null)
+                    {
+                        var translatedText = XElement.Parse(result).Value;
```

Comment:
Parse methods generally throw exception if the string is invalid. Check for null